source/aws_signing.c

```diff
@@ -1936,15 +2029,15 @@ static int s_add_authorization_to_result(
     struct aws_byte_cursor name;
     struct aws_byte_cursor value = aws_byte_cursor_from_buf(authorization_value);
 
-    if (state->config.signature_type == AWS_ST_HTTP_REQUEST_HEADERS) {
+    if (s_is_header_based_signature_value(state->config.signature_type)) {
         name = aws_byte_cursor_from_string(g_aws_signing_authorization_header_name);
         if (aws_signing_result_append_property_list(
                 &state->result, g_aws_http_headers_property_list_name, &name, &value)) {
             return AWS_OP_ERR;
         }
     }
 
-    if (state->config.signature_type == AWS_ST_HTTP_REQUEST_QUERY_PARAMS) {
+    if (s_is_query_param_based_signature_value(state->config.signature_type)) {
         name = aws_byte_cursor_from_string(g_aws_signing_authorization_query_param_name);
         if (aws_signing_result_append_property_list(
                 &state->result, g_aws_http_query_params_property_list_name, &name, &value)) {
```

Comment:
With this new way of signing a canonical request, should any headers/params (besides "Authorization") be added to the signing_result?

I ask because PR https://github.com/awslabs/aws-c-auth/pull/109 moved ALL that logic that adds new headers (except "Authorization") to `s_build_canonical_stable_header_list()`. When signing an existing canonical request, we'll skip that function, so we won't add anything but Authorization.

Is that how we want it to work? Are all the other headers required to be signed, and therefore must already be present? I guess X-Amz-Security-Token with omit_session_token is the only outlier, right?

Reply:
We don't want anything else added, just the signature computed.